tools/objc/libtool_check_unique.cc

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <regex>
+#include <fstream>
+#include <unordered_set>
+
+using namespace std;
+
+static unordered_set<string> sBasenames;
+
+string getBasename(string path) {
+    auto idx = path.find_last_of("/");
```

Comment:
not portable to windows, but is this libtool wrapper only used for macos?

Reply:
I believe this is only used on macOS, but I don't necessarily have visibility into what all other folks do with this. :-)

Reply:
i see. what would you suggest here? e.g., is there a way to detect if it's running on apple/linux, and only in that case apply this optimization?

Reply:
I honestly don't think it's worth worrying deeply about. Apple's libtool != GNU libtool and I doubt (though I could be wrong, our users are very clever) that anyone is cross-compiling their libtool onto Windows.

Plus there's lots of other stuff that the shell script already invoked that likely doesn't exist (e.g. xcrun).

I would add a comment here and move on.

If you want to check for platform and throw a helpful warning, it should be easy enough to do in the shell script, but I wouldn't bother here in the C++.

Reply:
good point about xcrun, yeah i'll just add a comment here